### PR TITLE
fix map [string] interface{}  is nil

### PR DESCRIPTION
**Pull request by lian-yue**
```go
	boolVal := false
	if out, err = expr.Eval("f != nil", map[string]interface{}{"f": boolVal}); err != nil {
		panic(err)
	}
	fmt.Println("out", out)


	trueVal := false
	if out, err = expr.Eval("f == true", map[string]interface{}{"f": trueVal}); err != nil {
		panic(err)
	}
	fmt.Println("out", out)


	stringVal := "string"
	if out, err = expr.Eval(`s == "string"`, map[string]interface{}{"s": stringVal}); err != nil {
		panic(err)
	}
	fmt.Println("out", out)


	var timeNil *time.Time
	if out, err = expr.Eval("t == nil", map[string]interface{}{"t": timeNil}); err != nil {
		panic(err)
	}
	fmt.Println("out", out)
```

### CODE REVIEW

File: vm/helpers.go [hunk 1]
```diff
@@ -332,15 +332,31 @@ func equal(a, b interface{}) interface{} {
...
 	case string:
-		return x == b.(string)
+		switch y := b.(type) {
+		case string:
+			return x == y
+		}
 	}
 	// Two nil values should be considered as equal.
-	if (a == nil || reflect.ValueOf(a).IsNil()) && (b == nil || reflect.ValueOf(b).IsNil()) {
+	if isNil(a) && isNil(b) {
```
Comment (antonmedv) [inline vm/helpers.go]: This should go into generate helper.

Comment (lian-yue) [reply vm/helpers.go]: Added to generate helper

**Comment by lian-yue** [comment]
Added to generate helper

File: vm/generate/main.go [hunk 2]
```diff
@@ -137,6 +140,20 @@ func main() {
 		echo(``)
 	}
 
+	// isNil func
+	echo(`func isNil(v interface{}) bool {`)
```
Comment (antonmedv) [inline vm/generate/main.go]: Better put it in runtime.go file.

**Comment by lian-yue** [comment]
ok

**Comment by antonmedv** [comment]
Fixed in 96a4cee7a22c684d5dbe44fdb0339f3270ce884d

**Pull request closed without merging**